lib/sensors/temperature.cpp

```diff
@@ -0,0 +1,53 @@
+#include "temperature.hpp"
+
+namespace hyped::sensors {
+
+Temperature::Temperature(hyped::io::I2c &i2c, hyped::core::ILogger &log) : log_(log), i2c_(i2c)
```

Comment:
we should take in a `mux_address` from the constructor here and use that as the "device address" when we read/write